test/exmg-snackbars.spec.ts

```diff
@@ -0,0 +1,55 @@
+import {showSnackBar} from '../exmg-snackbars';
+
+const {assert} = chai;
+
+suite('<exmg-snackbars>', function () {
+  suite('base usage', function () {
+    test('normal toast is displayed', function (done) {
+      showSnackBar('test message', {duration: 1000});
+
+      assert.exists(document.querySelector('#paper-toast-container'), 'Toast container should exist');
+      assert.exists(document.querySelector('paper-toast'), 'Toast should exist');
+      assert.notExists(document.querySelector('paper-icon-button'), 'Toast close button should exist');
+
+      setTimeout(() => {
+        assert.notExists(document.querySelector('paper-toast'), 'Toast should not exist');
+        done();
+      },  1000);
+    });
+
+    test('normal toast with custom node id is displayed', function (done) {
+      showSnackBar('test message', {duration: 1000, toastContainerNodeId: 'custom-id'});
+
+      assert.exists(document.querySelector('#' + 'custom-id'), 'Toast container should exist');
+      assert.exists(document.querySelector('paper-toast'), 'Toast should exist');
+      assert.notExists(document.querySelector('paper-icon-button'), 'Toast close button should exist');
+
+      setTimeout(() => {
+        assert.notExists(document.querySelector('paper-toast'), 'Toast should not exist');
+        done();
+      },  1000);
+    });
+
+    test('infinite toast with close button is displayed', function (done) {
+      showSnackBar('test message', {duration: 0, showCloseButton: true});
+
+      assert.exists(document.querySelector('#paper-toast-container'), 'Toast container should exist');
+      assert.exists(document.querySelector('paper-toast'), 'Toast should exist');
+      assert.exists(document.querySelector('paper-icon-button'), 'Toast close button should exist');
+
+      setTimeout(() => {
```

Comment:
instead nesting setTimeout you can improve readability 
```
const wait = (time: number) => new Promise(resolve => setTimeout(resolve, time));

test('test', async function() {
  this.timeout(5000); // tell test runner how long test may last
   // do some assertion
  await wait(100);
  // do some assertion
  await wait(4000);
// do some assertion
})
```

Reply:
done